habitat_sim/agent/agent.py

```diff
@@ -133,19 +132,16 @@ class Agent(object):
     """
 
     agent_config: AgentConfiguration
-    _sensors: SensorSuite
     controls: ObjectControls
     body: mn.scenegraph.AbstractFeature3D
 
     def __init__(
         self,
         scene_node: hsim.SceneNode,
         agent_config: Optional[AgentConfiguration] = None,
-        _sensors: Optional[SensorSuite] = None,
         controls: Optional[ObjectControls] = None,
     ) -> None:
         self.agent_config = agent_config if agent_config else AgentConfiguration()
-        self._sensors = _sensors if _sensors else SensorSuite()
```

Comment:
- We can keep the concept `_sensors` in the agent class, a convenient alias to the sensorSuite in the node. 
- For an atomic agent, which is the current case, ONLY one `_sensors` is needed and it serves as a reference to the `subtreeSensorSuite` of that node. You should do proper assignment here: `self._sensors = call getter function to obtain the subtreeSensorSuite of that node`
- For an articulated agent, which will be the case in the future, there would be two references, one is `_node_owned_sensors` and `_subtree_sensors` (for example, at the root node of this articulated agent, the `_subtree_sensors` means all the sensors being attached to this agent and the `_node_owned_sensors` are sensors on the "head" assuming the head node is the root node of this articulated agent.)